dtos/reading.go

```diff
@@ -696,9 +696,19 @@ func (b *BaseReading) Unmarshal(data []byte, unmarshal func([]byte, any) error)
 	b.Units = aux.Units
 	b.Tags = aux.Tags
 	b.BinaryReading = aux.BinaryReading
-	if aux.Value != nil {
-		b.SimpleReading = SimpleReading{Value: fmt.Sprintf("%s", aux.Value)}
+
+	// reading.value only support string for adding the reading
+	// numeric, numeric array are used for querying
+	//   for example, the user query event reading via event or reading http client with the parameter numeric=true, the http client can decode numeric into NumericReading
+	switch v := aux.Value.(type) {
+	case string: // string, for JSON strings
+		b.Value = v
+	case float64: // float64, for JSON numbers
```

Comment:
Add more explanation in comment to describe why only deal with float64 here

Reply:
Updated according to Cloud's comment. The json lib treats all number type as float64, so we don't need to handle other types.